rust/parquet/src/util/cursor.rs

```diff
@@ -15,20 +15,31 @@
 // specific language governing permissions and limitations
 // under the License.
 
-use std::cmp;
-use std::io::{self, Error, ErrorKind, Read};
+use std::io::{self, Error, ErrorKind, Read, Seek, SeekFrom};
 use std::rc::Rc;
+use std::{cmp, fmt};
 
 /// This is object to use if your file is already in memory.
 /// The sliceable cursor is similar to std::io::Cursor, except that it makes it easy to create "cursor slices".
 /// To achieve this, it uses Rc instead of shared references. Indeed reference fields are painfull
 /// because the lack of Generic Associated Type implies that you would require complex lifetime propagation when
 /// returning such a cursor.
 pub struct SliceableCursor {
-    inner: Rc<Vec<u8>>,
     start: u64,
     length: usize,
     pos: u64,
+    inner: Rc<Vec<u8>>,
+}
+
+impl fmt::Debug for SliceableCursor {
```

Comment:
I used this formulation rather than `[#derive(Debug)]` to avoid getting the buffer contents dumped which can be lots of data to sort through.